xncml/core.py

```diff
@@ -278,3 +286,127 @@ def to_ncml(self, path=None):
+        `groups` key. Additional keys prefixed with `@` may be included for <netcdf> tag attributes,
+        for example `@location`.
+
+        References
+        ----------
+        http://cf-json.org/specification
+        """
+        res = OrderedDict()
+        nc = self.ncroot['netcdf']
+
+        for key, val in nc.items():
+            if key[0] == '@':
+                res[key] = val
+            if key == 'dimension':
+                res.update(_dims_to_json(val))
+            if key == 'group':
+                res.update(_groups_to_json(val))
+            if key == 'attribute':
+                res.update(_attributes_to_json(val))
+            if key == 'variable':
+                res.update(_variables_to_json(val))
+
+        return res
+
+
+def _dims_to_json(dims: list) -> dict:
+    """The dimensions object has dimension id:size as its key:value members."""
+    out = OrderedDict()
+    for dim in dims:
+        if int(dim['@length']) > 1:
+            out[dim['@name']] = int(dim['@length'])
+
+    return {'dimensions': out}
+
+
+def _groups_to_json(groups: list) -> dict:
+    out = OrderedDict()
+    for group in groups:
+        name = group['@name']
+        out[name] = OrderedDict()
+        if 'attribute' in group:
+            out[name].update(_attributes_to_json(group['attribute']))
+        if 'group' in group:
+            out[name].update(_groups_to_json(group['group']))
+
+    return {'groups': out}
+
+
+def _attributes_to_json(attrs: list) -> dict:
+    """The attributes object contains arbitrary attributes as its key:value members."""
+    SPECIAL_ATTRS = ['missing_value', 'cell_methods']
+
+    out = OrderedDict()
+    for attr in attrs:
+        if attr['@name'] not in SPECIAL_ATTRS:
+            try:
+                out[attr['@name']] = _cast(attr)
+            except ValueError as exc:
+                warn(f"Could not cast {attr['@name']}:\n{exc}")
+
+    return {'attributes': out}
+
+
+def _variables_to_json(variables: list) -> dict:
+    """The variables definition object has variable id:object as its key:value members.
+
+    Each variable object MUST include shape, attributes and data objects.
+    The shape field is an array of dimension IDs which correspond to the array ordering of the variable data.
+    """
+    AXIS_VAR = ['time', 'lat', 'latitude', 'lon', 'longitude', 'site']
+
+    out = OrderedDict()
+
+    # Put axis variables first
+    for special_var in AXIS_VAR:
+        if special_var in [v['@name'] for v in variables]:
+            out[special_var] = None
+
+    for var in variables:
+        name = var['@name']
+        out[name] = OrderedDict()
+
+        if '@shape' in var:
+            out[name]['shape'] = var['@shape'].split(' ')
+
+        if '@type' in var:
+            out[name]['type'] = var['@type']
+
+        if 'attribute' in var:
+            out[name].update(_attributes_to_json(var['attribute']))
+
+        if 'values' in var:
+            out[name]['data'] = _cast(var)
```

Comment:
Didn't the docstring say that `data` was not included?

Reply:
My bad, will fix.